src/LagrangianParticleTracking/calculate_particle_tendency_contributions.jl

```diff
@@ -0,0 +1,84 @@
+function calculate_particle_tendency_contributions!(particles, model, active_properties)
+    arch = model.architecture
+
+    workgroup = min(length(particles), 256)
+    worksize = length(particles)
+    
+    Gp_kernel! = calculate_particle_tendency_kernel!(device(arch), workgroup, worksize)
+
+    events = []
+
+    for field in active_properties
+        property = getproperty(particles.properties, field.property)
+        tendency = model.timestepper.Gⁿ[field.tracer]
+
+        Gp_event = Gp_kernel!(property, tendency, particles, model.grid, dependencies = Event(device(arch)))
+
+        push!(events, Gp_event)
+    end
+
+    wait(device(arch), MultiEvent(Tuple(events)))
+
+    return nothing
+end
+
+calculate_particle_tendency_contributions!(model) = calculate_particle_tendency_contributions!(model.particles, model)
+calculate_particle_tendency_contributions!(particles::Nothing, model) = nothing
+calculate_particle_tendency_contributions!(particles, model) = !isnothing(particles.parameters) ? (:active_properties in keys(particles.parameters) ? calculate_particle_tendency_contributions!(particles, model, particles.parameters.active_properties) : nothing) : nothing
```

Comment:
Trying to understand this: your proposed API asks users to add a property `active_properties` to the particle parameters? If they don't add it, then due to this `if`-statement, nothing happens?

Reply:
Would another solution add a new property to `LagrangianParticles`? This would allow us to use dispatch at a higher level rather than these nested if statements.

Reply:
Adding a new property probably would be the most straight forward